frank/fit.py

```diff
@@ -0,0 +1,275 @@
+    """
+
+    logging.info('  Determining disc geometry')
+
+    if not model['geometry']['fit_geometry']:
+        geom = frank.geometry.FixedGeometry(0., 0., 0., 0.)
+
+    else:
+        if model['geometry']['known_geometry']:
+            geom = frank.geometry.FixedGeometry(model['geometry']['inc'],
+                                                model['geometry']['pa'],
+                                                model['geometry']['dra'],
+                                                model['geometry']['ddec']
+                                               )
+
+        else:
+            if model['geometry']['fit_phase_offset']:
+                geom = frank.geometry.FitGeometryGaussian()
+
+            else:
+                geom = frank.geometry.FitGeometryGaussian(
+                                        phase_centre=(model['geometry']['dra'],
+                                        model['geometry']['ddec'])
+                                        )
+
+            t1 = time.time()
+            geom.fit(u, v, vis, weights)
+            logging.info('    Time taken to fit geometry %.1f sec'%(time.time() - t1))
+
+    logging.info('    Using: inc  = %.2f deg,\n           PA   = %.2f deg,\n'
+          '           dRA  = %.2e arcsec,\n           dDec = %.2e arcsec'
+          %(geom.inc, geom.PA, geom.dRA, geom.dDec))
+
+    return geom
+
+
+def perform_fit(model, u, v, vis, weights, geom):
+    """
+    Deproject the observed visibilities and fit them for the brightness profile.
+
+    Parameters
+    ----------
+    model : dict
+          Dictionary containing model parameters the fit uses
+    u, v : array, unit = :math:`\\lambda`
+          u and v coordinates of observations
+    vis : array, unit = Jy
+          Real component of observed visibilities
+    weights : array, unit = Jy^-2
+          Weights assigned to observed visibilities, of the form
+          :math:`1 / \\sigma^2`
+    geom : SourceGeometry object
+          Fitted geometry (see frank.geometry.SourceGeometry)
+
+    Returns
+    -------
+    sol : _HankelRegressor object
+          Reconstructed profile using Maximum a posteriori power spectrum
+          (see frank.radial_fitters.FrankFitter)
+    """
+
+    logging.info('  Fitting for brightness profile')
+
+    FF = frank.radial_fitters.FrankFitter(Rmax=model['hyperpriors']['rout'],
+                     N=model['hyperpriors']['n'],
+                     geometry=geom,
+                     alpha=model['hyperpriors']['alpha'],
+                     weights_smooth=model['hyperpriors']['wsmooth']
+                     )
+
+    t1 = time.time()
+    sol = FF.fit(u, v, vis, weights)
+    logging.info('    Time taken to fit profile (with %.0e visibilities and %s'
+          ' collocation points) %.1f sec'%(len(vis), model['hyperpriors']['n'],
+          time.time() - t1))
+
+    return sol, FF.iteration_diagnostics
+
+
+def output_results(model, u, v, vis, weights, geom, sol, iteration_diagnostics,
```

Comment:
You shouldn't require iteration_diagnostics if you don't want to plot them. Use iteration_diagnostics=None to determine specify that they shouldn't be plotted